client/state/themes/selectors/is-theme-purchased.ts

```diff
@@ -0,0 +1,27 @@
+import { getSitePurchases } from 'calypso/state/purchases/selectors';
+import type { AppState } from 'calypso/types';
+
+import 'calypso/state/themes/init';
+
+/**
+ * Returns whether the theme has been purchased for the given site. Note that this only works for
+ * premium themes that are added to A8c repos -- marketplace themes need to be checked via a different approach.
+ *
+ * Use this selector alongside with the <QuerySitePurchases /> component.
+ *
+ * @param {AppState} state Global state tree
```

Comment:
Nitpick, would be nice to neatly align the params like the original does for readability.

Reply:
Done in 140ea35.